grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -426,7 +428,22 @@ private void executePrepared(
           .whenComplete(
               (result, t) -> {
                 if (t != null) {
-                  handleException(t, responseObserver);
+                  if (t instanceof PreparedQueryNotFoundException) {
+                    preparedCache.invalidate(prepareInfo);
+
+                    prepareQuery(connection, prepareInfo, parameters.getTracing())
+                        .whenComplete(
+                            (p, t1) -> {
+                              if (t1 != null) {
+                                handleException(t1, responseObserver);
+                              } else {
+                                executePrepared(
```

Comment:
Should we be checking for infinite loops here? It's probably not likely, but still... :thinking:

Reply:
I thought about that but noticed `PersistenceBackedDataStore.java` doesn't check so I figured we'd be fine without it.

Reply:
I do believe the same problem exists there too, but I think it is not likely to result in real issues. The schema would have to change very often or the statement preparation has to have bugs for this infinite loop to manifest. +0

Reply:
Maybe we should do an exponential backoff before retry?